torch/distributed/launch.py

```diff
@@ -174,10 +174,10 @@ def parse_args():
                              "the IP address or the hostname of node 0, for "
                              "single node multi-proc training, the "
                              "--master_addr can simply be 127.0.0.1")
-    parser.add_argument("--master_port", default=29500, type=int,
+    parser.add_argument("--master_port", default=0, type=int,
```

Comment:
Can we keep the default the same to avoid breaking any existing applications? We can extend the docs to mention that if you pass in 0, we will find an appropriate free port for you.

This way users that would like to take advantage of this can just pass in 0 for master port.

Reply:
That sounds good.